src/olympia/migrations/947-delete-orphaned-validation-jobs-add-fks.sql

```diff
@@ -0,0 +1,61 @@
+-- Delete orphaned validation jobs (since we removed the 'Mobile' app).
+DELETE FROM validation_job WHERE curr_max_version_id NOT IN (SELECT id FROM appversions);
+DELETE FROM validation_job WHERE target_version_id NOT IN (SELECT id FROM appversions);
+
+/*
+For validation_job.curr_max_version_id, validation_job.target_version_id and validation_job.creator_id:
+ 1. we check whether FK constraints exist and drop them if so (in order to do the next step).
+ 2. we check whether indices exist and drop them if so (because their names have not been generated by django).
+ 3. we add FK constraints with proper names generated by django.
+*/
+
+-- Delete existing foreign key constraints so we can delete the keys afterwards.
+SET @FK_CURR_MAX_VERSION_ID := (SELECT CONSTRAINT_NAME FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE TABLE_SCHEMA = 'olympia' AND TABLE_NAME = 'validation_job' AND COLUMN_NAME = 'curr_max_version_id');
+
+SET @QUERY_DROP_FK_CURR_MAX_VERSION_ID = IF(@FK_CURR_MAX_VERSION_ID, CONCAT('ALTER TABLE validation_job DROP FOREIGN KEY ', @FK_CURR_MAX_VERSION_ID, ';'), 'SELECT 0;');
```

Comment:
Depending on when your database was created, it might or might not have FKs or regular indices.